src/core/execute.c

```diff
@@ -3963,10 +4038,15 @@ void exec_context_dump(const ExecContext *c, FILE* f, const char *prefix) {
         fprintf(f, "%sRuntimeDirectoryPreserve: %s\n", prefix, exec_preserve_mode_to_string(c->runtime_directory_preserve_mode));
 
         for (dt = 0; dt < _EXEC_DIRECTORY_TYPE_MAX; dt++) {
+                size_t j;
+
                 fprintf(f, "%s%sMode: %04o\n", prefix, exec_directory_type_to_string(dt), c->directories[dt].mode);
 
-                STRV_FOREACH(d, c->directories[dt].paths)
-                        fprintf(f, "%s%s: %s\n", prefix, exec_directory_type_to_string(dt), *d);
+                for (j = 0; j < c->directories[dt].n_directories; j++)
+                        fprintf(f, "%s%s%s: %s\n", prefix,
+                                exec_directory_type_to_string(dt),
+                                c->directories[dt].directories[j].ignore ? "-" : "",
+                                c->directories[dt].directories[j].path);
```

Comment:
This gives something like "RuntimeDirectory-: foobar" (the dash is in the wrong place).

Reply:
ouch. fixed.